packages/plugin-background-removal-web/README.md

```diff
@@ -71,3 +75,38 @@ For optimal performance using the correct CORS headers is important. See the lib
 'Cross-Origin-Opener-Policy': 'same-origin',
 'Cross-Origin-Embedder-Policy': 'require-corp'
 ```
+
+## Custom Background Removal Provider
+
+It is possible to declare a different provider for the background removal process.
+
+```typescript
+BackgroundRemovalPlugin({
+    provider: {
+        type: 'custom',
+        // This is an optional function that can do some pre-processing, e.g. creating
+        // a mask that is applied on all images from the source set in the process function.
+        // The returned blob is passed as the second argument in `process`
+        preprocess: async (uriToProcess: string) => {
+            const mask = await segmentForeground(uriToProcess);
+            return mask;
+        },
+        process: (
+            sourceSet: {
+                uri: string;
+                width?: number;
+                height?: number;
+            }[],
+            preprocessedData?: Blob
+        ): Blob[] => {
+            const masked = await Promise.all(
+                sourceSet.map((source) => {
+                    // ...
+                })
+            );
+
+            return masked;
+        }
```

Comment:
I do not really understand the distinction between preprocess and process as well as I'm unsure about what uriToProcess should relate to. Which URI is passed into it? Why is it not possible to move both functions into one? 
From glancing over the docs I'm also not sure why the process just returns a list of blobs and how they relate to any existing data or if e.g the order of the blobs matters.

Reply:
Removed the preprocess step and adapted the README [here](https://github.com/imgly/plugins/pull/18/commits/ab7c64d85d97c85b1577fc05853505c401adb39a).

Reply:
Well done, looks much cleaner now! 
Just for my understanding, why are we returning blobs instead of e.g "uri"s from the processing call? Are the blobs then put into the configured `onUpload` handler?

Reply:
Yes!